File: FinTrack/Models/Account.cs
namespace FinTrack.Models;

public class Account
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;
    public string Name { get; set; } = null!;
    public AccountType Type { get; set; }
    public string CurrencyCode { get; set; } = null!;
    public decimal OpeningBalance { get; set; }
    public bool IsArchived { get; set; }
    public DateTime CreationTimeAtUtc { get; set; }
    public DateTime UpdateTimeAtUtc { get; set; }
    public Guid? SavingsGoalId { get; set; }
    public SavingsGoal? SavingsGoal { get; set; }

    public ICollection<Transaction> Transactions { get; set; } = [];
}

public enum AccountType : short
{
    Cash = 1,
    DebitCard = 2,
    Savings = 3,
    Deposit = 4,
    Other = 5
}

File: FinTrack.Client/src/types/account.ts
export interface CreateAccountRequest {
    name: string;
    type: number;
    currencyCode: string;
    openingBalance: number;
}

export interface UpdateAccountRequest {
    name?: string;
    type?: AccountType;
}

export interface Account {
    id: string;
    name: string;
    type: AccountType;
    currencyCode: string;
    balance: number;
    isArchived: boolean;
    creationTimeAtUtc: string;
    updateTimeAtUtc: string;
}

export enum AccountType {
    Cash = 1,
    DebitCard = 2,
    Savings = 3,
    Deposit = 4,
    Other = 5
}

export const accountTypeLabels: Record<AccountType, string> = {
    [AccountType.Cash]: "Наличные",
    [AccountType.DebitCard]: "Карта",
    [AccountType.Savings]: "Накопительный счёт",
    [AccountType.Deposit]: "Вклад",
    [AccountType.Other]: "Другое"
};

export type AccountOption = {
    value: string;
    label: string;
};
